crates/valence_inventory/src/lib.rs

```diff
@@ -1235,11 +1235,11 @@ fn handle_update_selected_slot(
                     // The client is trying to interact with a slot that does not exist, ignore.
                     continue;
                 }
-                held.held_item_slot = convert_hotbar_slot_id(pkt.slot as u16);
+                held.held_item_slot = convert_hotbar_slot_id(pkt.slot);
 
                 events.send(UpdateSelectedSlotEvent {
                     client: packet.client,
-                    slot: pkt.slot,
+                    slot: pkt.slot as u8,
                 });
```

Comment:
These changes seem unrelated to this PR. If it's not relevant to boss bars, revert them.